src/pages/OnboardingAccounting/BaseOnboardingAccounting.tsx

```diff
@@ -0,0 +1,190 @@
+import Text from '@components/Text';
+import useLocalize from '@hooks/useLocalize';
+import useResponsiveLayout from '@hooks/useResponsiveLayout';
+import useStyleUtils from '@hooks/useStyleUtils';
+import useTheme from '@hooks/useTheme';
+import useThemeStyles from '@hooks/useThemeStyles';
+import Navigation from '@libs/Navigation/Navigation';
+import variables from '@styles/variables';
+import * as Report from '@userActions/Report';
+import * as Welcome from '@userActions/Welcome';
+import * as OnboardingFlow from '@userActions/Welcome/OnboardingFlow';
+import CONST from '@src/CONST';
+import type {OnboardingAccountingType} from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {} from '@src/types/onyx/Bank';
+import type {BaseOnboardingAccountingProps} from './types';
+
+type OnboardingListItem = ListItem & {
+    keyForList: OnboardingAccountingType;
+};
+
+function BaseOnboardingAccounting({shouldUseNativeStyles, route}: BaseOnboardingAccountingProps) {
+    const styles = useThemeStyles();
+    const theme = useTheme();
+    const StyleUtils = useStyleUtils();
+    const {translate} = useLocalize();
+    const {onboardingIsMediumOrLargerScreenWidth, shouldUseNarrowLayout} = useResponsiveLayout();
+    const [onboardingPurposeSelected] = useOnyx(ONYXKEYS.ONBOARDING_PURPOSE_SELECTED);
+    const [onboardingPolicyID] = useOnyx(ONYXKEYS.ONBOARDING_POLICY_ID);
+    const [onboardingAdminsChatReportID] = useOnyx(ONYXKEYS.ONBOARDING_ADMINS_CHAT_REPORT_ID);
+    const [onboardingCompanySize] = useOnyx(ONYXKEYS.ONBOARDING_COMPANY_SIZE);
+
+    const [userReportedIntegration, setUserReportedIntegration] = useState<OnboardingAccountingType | undefined>(undefined);
+    const [error, setError] = useState('');
+
+    const accountingOptions: OnboardingListItem[] = useMemo(() => {
+        const policyAccountingOptions = Object.values(CONST.POLICY.CONNECTIONS.NAME).map((connectionName): OnboardingListItem => {
+            let text;
+            let accountingIcon;
+            switch (connectionName) {
+                case CONST.POLICY.CONNECTIONS.NAME.QBO: {
+                    text = translate('workspace.accounting.qbo');
+                    accountingIcon = Expensicons.QBOCircle;
+                    break;
+                }
+                case CONST.POLICY.CONNECTIONS.NAME.XERO: {
+                    text = translate('workspace.accounting.xero');
+                    accountingIcon = Expensicons.XeroCircle;
+                    break;
+                }
+                case CONST.POLICY.CONNECTIONS.NAME.NETSUITE: {
+                    text = translate('workspace.accounting.netsuite');
+                    accountingIcon = Expensicons.NetSuiteSquare;
+                    break;
+                }
+                default: {
+                    text = translate('workspace.accounting.intacct');
+                    accountingIcon = Expensicons.IntacctSquare;
+                    break;
+                }
```

Comment:
Let's not default to any. There is no default. Also this is causing this bug

<img width="468" alt="Screenshot 2024-10-08 at 11 04 06 AM" src="https://github.com/user-attachments/assets/afdb8174-22c2-4a62-bea5-06684aeb10da">

Reply:
Let's also log a warning if we reach the default case

Reply:
@s77rt This happens because we added a new connect is `quickbooksDesktop` in the const, I think we only need to not default any, no need to add a log warning.